### DISCUSSION TITLE
What caddy is doing here? Documentation missing.

### DISCUSSION THREAD

**Discussion by afzl-wtu** [ZoeyVid/NPMplus#370] [Q&A] [answered]
I was looking for updates in your repo and come across seeing caddy. What is caddy doing here. I could not find any documentation about caddy

**Comment by Zoey2936** [comment]
see here: https://github.com/ZoeyVid/nginx-proxy-manager/blob/develop/compose.override.yaml I use it to redirect all HTTP to HTTPS

**Reply by afzl-wtu** [reply]
Doesn't it slow down using two webserver in container? Network request first goes to caddy and then come to nginx.

**Reply by Zoey2936** [reply]
no, this is an optional thing, if you don't use the compose.override.yaml, caddy is at no place used

**Reply by Zoey2936** [reply, marked as answer]
by default everything is done using nginx, but the compose.override.yaml disabled http in nginx, so nginx only handels https and caddy will only do http to https redirects

**Reply by gregfr** [reply]
I'm curious, why did you choose to do that?

**Reply by Zoey2936** [reply]
Thats is because in the past nginx didn't supported h2c (http2 on http), but caddy did

**Reply by gregfr** [reply]
Thanks for your answer :) "in the past" means it does now?

**Reply by Zoey2936** [reply]
yes
